cmd/create/cluster/cmd.go

```diff
@@ -515,6 +527,14 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	isSubnetIDsSet := cmd.Flags().Changed("subnet-ids")
+	isAvailabilityZonesSet := cmd.Flags().Changed("availability-zones")
+	// Setting subnet IDs is choosing BYOVPC implicitly,
+	// and selecting availability zones is only allowed for non-BYOVPC clusters
+	if isSubnetIDsSet && isAvailabilityZonesSet {
```

Comment:
I wasn't sure about this constraint. 
In CS we require both **subnets** and **availability zones** for BYOVPC cluster.

Here, if providing both, the alternative will be to validate that the subnets belonging to the availability zones provided.

If we won't allow providing both, we can prevent some user errors, as ROSA CLI is figuring out the availability zones for the subnets provided https://github.com/openshift/rosa/blob/master/cmd/create/cluster/cmd.go#L1277.

Reply:
is this a question ?